### PR TITLE
[Clock] Document createFromTimestamp() method

### LINKED ISSUE #19766: [Clock] Add a polyfill for DateTimeImmutable::createFromTimestamp()
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#54442
| PR author(s) | @derrabus
| Merged in    | 7.1

We created this issue to not forget to document this new feature. We would really appreciate if you can help us with this task. If you are not sure how to do it, please ask us and we will help you.

To fix this issue, please create a PR against the 7.1 branch in the [symfony-docs repository](https://github.com/symfony/symfony-docs).

Thank you! :smiley:

### PR DESCRIPTION
**Pull request by javiereguiluz**
Fixes #19766

### CODE REVIEW

File: components/clock.rst [hunk 1]
```diff
@@ -229,6 +229,21 @@ The constructor also allows setting a timezone or custom referenced date::
     $referenceDate = new \DateTimeImmutable();
     $relativeDate = new DatePoint('+1month', reference: $referenceDate);
 
+The ``DatePoint`` class also provides a named constructor to create dates from
+timestamps::
+
+    $dateOfFirstCommitOfSymfonyProject = DatePoint::createFromTimestamp('1129645656');
```
Comment (xabbuh) [inline components/clock.rst]:
```suggestion
    $dateOfFirstCommitOfSymfonyProject = DatePoint::createFromTimestamp(1129645656);
```

File: components/clock.rst [hunk 2]
```diff
@@ -229,6 +229,21 @@ The constructor also allows setting a timezone or custom referenced date::
...
+    // equivalent to:
+    // $dateOfFirstCommitOfSymfonyProject = (new \DateTimeImmutable())->setTimestamp('1129645656');
+
+    // negative timestamps (for dates before January 1, 1970) and fractional timestamps
+    // (for high precision sub-second datetimes) are also supported
+    $dateOfFirstMoonLanding('-14182940');
```
Comment (xabbuh) [inline components/clock.rst]:
```suggestion
    $dateOfFirstMoonLanding = DatePoint::createFromTimestamp(-14182940);
```

File: components/clock.rst [hunk 3]
```diff
@@ -229,6 +229,21 @@ The constructor also allows setting a timezone or custom referenced date::
     $referenceDate = new \DateTimeImmutable();
     $relativeDate = new DatePoint('+1month', reference: $referenceDate);
 
+The ``DatePoint`` class also provides a named constructor to create dates from
+timestamps::
+
+    $dateOfFirstCommitOfSymfonyProject = DatePoint::createFromTimestamp('1129645656');
+    // equivalent to:
+    // $dateOfFirstCommitOfSymfonyProject = (new \DateTimeImmutable())->setTimestamp('1129645656');
```
Comment (xabbuh) [inline components/clock.rst]:
```suggestion
    // $dateOfFirstCommitOfSymfonyProject = (new \DateTimeImmutable())->setTimestamp(1129645656);
```

**Comment by javiereguiluz** [comment]
@xabbuh thanks for your review and I'm sorry for these gross mistakes. I was in a rush and I committed without a proper review first. Everything is fixed now. Thanks!

**Review by xabbuh — APPROVED** [review]

**Comment by derrabus** [comment]
Thank you for your work on this, @javiereguiluz!

I think, it is worth mentioning that this method is part of a set of (more undocumented) polyfills. The `createFromTimestamp()` method will be added to `DateTimeImmutable` (and thus to our `DatePoint` class as well) in PHP 8.4. My polyfill makes this functionality available on PHP 8.2 already. The other polyfilled methods that have existed before my PR already are `setMicrosecond()` and `getMicrosecond()`, both added by @nicolas-grekas in symfony/symfony#53942.

**Pull request merged**